src/Gaffer/ValuePlug.cpp

```diff
@@ -391,13 +391,17 @@ class ValuePlug::HashProcess : public Process
 
 		static void dirtyLegacyCache()
 		{
-			g_legacyGlobalDirtyCount++;
-		
-	
-			if( g_legacyGlobalDirtyCount >= DIRTY_COUNT_RANGE_HALF )
+			if( g_hashCacheMode != HashCacheMode::Standard )
 			{
-				g_legacyGlobalDirtyCount = 0;
-				clearCache();
+				uint64_t c = g_legacyGlobalDirtyCount;
+				while( !g_legacyGlobalDirtyCount.compare_exchange_weak( c, ( c + 1 ) % DIRTY_COUNT_RANGE_HALF ) )
+				{
+				}
+		
+				if( c == DIRTY_COUNT_RANGE_HALF - 1 )
+				{
+					clearCache();
+				}
```

Comment:
I don't think this provides thread safety. Imagine two threads, each with their own node graph. One makes an edit and detects wraparound, then the other makes an edit and launches a compute _before_ the first thread has got to `clearCache()`.

Given that we don't really make use of different graphs in different threads (in the UI, all graphs are managed on the UI thread) it's debatable how much we should care about this. I suggest we either simplify things and make no pretencee of thread safety, or we use a mutex, knowing that we shouldn't hit it often because we shouldn't be in Legacy mode, and even then contention with another thread is unlikely.

Reply:
Implemented using spin mutex as of a82db4b